doc/programming/language/statements.rst

```diff
@@ -41,6 +41,19 @@ will be carried along with the exception.
 
 .. include:: /autogen/types/generic-begin.rst
 
+.. _statement_block:
+
+``block``
+---------
+
+::
+
+    { STATEMENTS }
+
+Defines a scope that limits the visibility of local variables to within
+the block. This also allows variable shadowing, meaning inner variables
+can temporarily override outer variables or constants with the same name.
+
```

Comment:
Documenting this at all feel confusing to me, at least here, e.g., one might wonder whether Spicy blocks are statements like for [Rust's block expressions](https://doc.rust-lang.org/reference/expressions/block-expr.html)). We say somewhere that Spicy has a C-like syntax, and these are just scopes.

We could probably just not document this at all.